monailabel/endpoints/datastore.py

```diff
@@ -181,6 +182,28 @@ def update_label_info(label: str, tag: str, info: str = Form("{}"), user: Option
     return instance.datastore().update_label_info(label, tag, i)
 
 
+def download_dataset(limit_cases: Optional[int] = None):
+    instance: MONAILabelApp = app_instance()
+    dl = instance.datastore().datalist()
+
+    if not len(dl) > 0:
+        return {}
+
+    if limit_cases and limit_cases in list(range(1, len(dl))):
+        logger.info(f"Number of cases in datalist reduced to: {limit_cases} of {len(dl)} case(s)")
+        dl = dl[:limit_cases]
+
+    with zipfile.ZipFile("dataset.zip", mode="w") as archive:
+        logger.info(f"ZIP archive will be written to: {archive.filename}")
+        for d in dl:
+            # write image and corresponding label file to archive
+            for key in d.keys():
+                path = d[key]
+                archive.write(path, arcname=os.path.join(key, os.path.basename(path)))
```

Comment:
Better we define an abstract method in dataset interface and let derived classes provide the implementation.. some support file.. some support http.. and some can throw NotSupported exception..